### DISCUSSION TITLE
Неправильное и лишнее логирование действий

### DISCUSSION THREAD

**Discussion by rovius** [LacunaHub/.github#13] [Общее]
В событиях журнала действий (событий сервера) иногда неправильно указывается автор действия, т.е. кто его совершил. Например, в сообщении Лакуны может быть указано, что действие сделал не я, а другой администратор сервера или вообще рандомный участник, и эта информация не будет сходиться с журналом аудита дискорда.

Кроме того, имеют место лишние события, то есть событие, которого не происходило или которое происходило очень давно. Яркий пример тому - добавление ролей, указываются роли давно добавленные или которые уже были у участника.

Патч 4.11.6 решил проблему, но лишь временно - со вчерашнего дня (20 апреля) снова появляются сообщения о нерабочих логах. В [ветке в дискорде](https://discord.com/channels/740586549145763960/1229505442737033358) поподробнее.

**Comment by danliyev** [comment]
Думаю, ограничить срабатывание события добавления/снятия роли на участников, которые не находятся в кэше. Они чаще всего генерируют неверную информацию в логах.

Лакуна берёт автора события в журнале из аудита в дискорде. С приходом онбординга, участники могут выдавать роли себе сами и они отображаются в аудите. Фикс под вопросом, надо смотреть.
